pgrx-pg-sys/src/submodules/ffi.rs

```diff
@@ -9,7 +9,64 @@
 //LICENSE Use of this source code is governed by the MIT license that can be found in the LICENSE file.
 #![deny(unsafe_op_in_unsafe_fn)]
 
-use cee_scape::SigJmpBufFields;
+#[cfg(not(all(
+    any(target_os = "linux", target_os = "macos"),
+    any(target_arch = "x86_64", target_arch = "aarch64")
+)))]
+mod cee_scape {
+    #[cfg(not(feature = "cshim"))]
+    compile_error!("target platform cannot work without feature cshim");
```

Comment:
Hm. Is there a reason we're not just using the `use_c_to_interface_with_setjmp` feature of `cee_scape`?

Reply:
1. To support more CPUs and Windows, we need a patch of `cee_scape`.
2. In `postgres/src/include/c.h`, sigsetjmp is defined as:
```c
/*
 * When there is no sigsetjmp, its functionality is provided by plain
 * setjmp.  We now support the case only on Windows.  However, it seems
 * that MinGW-64 has some longstanding issues in its setjmp support,
 * so on that toolchain we cheat and use gcc's builtins.
 */
#ifdef WIN32
#ifdef __MINGW64__
typedef intptr_t sigjmp_buf[5];
#define sigsetjmp(x,y) __builtin_setjmp(x)
#define siglongjmp __builtin_longjmp
#else							/* !__MINGW64__ */
#define sigjmp_buf jmp_buf
#define sigsetjmp(x,y) setjmp(x)
#define siglongjmp longjmp
#endif							/* __MINGW64__ */
#endif							/* WIN32 */
```
I feel it's better to include the header and use `sigsetjmp` in `pgrx-cshim.c` than adding a `__builtin_setjmp` version of `call_closure_with_setjmp` in `cee-scape` and gating the variants by cfg...

Reply:
Ah, interesting.